### PR TITLE
mm-added all school frontend

### PR DESCRIPTION
**Pull request by maya-maye**
In this PR, I copied all the school frontend files including tests, stories,  pages, etc.

### CODE REVIEW

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -81,11 +81,19 @@ export default function AppNavbar({ currentUser, systemInfo, doLogout, currentUr
             <Nav className="mr-auto">
               {
                 hasRole(currentUser, "ROLE_USER") && (
+
                   <NavDropdown title="Schools" id="appnavbar-schools-dropdown" data-testid="appnavbar-schools-dropdown" >
                     <NavDropdown.Item href="/schools/list" data-testid="appnavbar-schools-list">List</NavDropdown.Item>
                     {
                       hasRole(currentUser, "ROLE_ADMIN") && (
                         <NavDropdown.Item href="/schools/create" data-testid="appnavbar-schools-create">Create</NavDropdown.Item>
+
```
Comment (Arik) [inline frontend/src/main/components/Nav/AppNavbar.js]: It looks like the schools and laptop navigation got mashed together when they should be separate

File: frontend/src/main/pages/Schools/SchoolDetailsPage.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export default function SchoolDetailsPage() {
+  let { id } = useParams();
+
+  const response = schoolUtils.getById(id);
+
+  return (
+    <BasicLayout>
+      <div className="pt-2">
+        <h1>School Details</h1>
+        <SchoolTable schools={[response.school]} showButtons={false} />
```
Comment (Arik) [inline frontend/src/main/pages/Schools/SchoolDetailsPage.js:15]: Changing this to `[response.school || {}]` will make it work with the storybook template from ucsbdates

**Comment by Arik** [comment]
Also needs storybook entries for the pages (they aren't in team01, use ucsbdates as a reference)

**Comment by justinthecao** [comment]
i think i fixed the navbar?

File: frontend/src/main/components/Nav/AppNavbar.js (hunk 1 above)
Comment (Arik) [reply frontend/src/main/components/Nav/AppNavbar.js]: Resolved by 655658c35d711f396f89c8e8c7d32c69ed65956e

File: frontend/src/main/pages/Schools/SchoolCreatePage.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import axios from "axios";
+import AxiosMockAdapter from "axios-mock-adapter";
+export default function SchoolCreatePage() {
+
+  let navigate = useNavigate(); 
+
+  const onSubmit = async (school) => {
+    const createdSchool = schoolUtils.add(school);
+    console.log("createdSchool: " + JSON.stringify(createdSchool));
+    navigate("/schools");
```
Comment (Arik) [inline frontend/src/main/pages/Schools/SchoolCreatePage.js]: navigate("/schools/list");

File: frontend/src/main/pages/Schools/SchoolEditPage.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    let { id } = useParams();
+
+    let navigate = useNavigate(); 
+
+    const response = schoolUtils.getById(id);
+
+    const onSubmit = async (school) => {
+        const updatedSchool = schoolUtils.update(school);
+        console.log("updatedSchool: " + JSON.stringify(updatedSchool));
+        navigate("/schools");
```
Comment (Arik) [inline frontend/src/main/pages/Schools/SchoolEditPage.js]: navigate("/schools/list");

File: frontend/src/main/pages/Schools/SchoolIndexPage.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    const schoolCollection = schoolUtils.get();
+    const schools = schoolCollection.schools;
+
+    const showCell = (cell) => JSON.stringify(cell.row.values);
+
+    const deleteCallback = async (cell) => {
+        console.log(`SchoolIndexPage deleteCallback: ${showCell(cell)})`);
+        schoolUtils.del(cell.row.values.id);
+        navigate("/schools");
```
Comment (Arik) [inline frontend/src/main/pages/Schools/SchoolIndexPage.js]: navigate("/schools/list");

File: frontend/src/stories/pages/Schools/SchoolDetailsPage.stories.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import React from 'react';
+
+import SchoolDetailsPage from "main/pages/Schools/SchoolDetailsPage";
+
+export default {
+    title: 'pages/Schools/SchoolDetailsPage',
+    component: SchoolDetailsPage
+};
+
+const Template = () => <SchoolDetailsPage />;  // TODO: Does not work - needs id param
```
Comment (Arik) [inline frontend/src/stories/pages/Schools/SchoolDetailsPage.stories.js:11]: const Template = () => <SchoolDetailsPage />;

**Pull request merged**